app/controllers/participants_controller.rb

```diff
@@ -6,13 +6,32 @@ def index
   end
 
   def create
-    @participant = Participant.find_by_email(participant_params)
+    
+    @participant = Participant.find_by_email(params[:participant][:email])
     if @participant == nil
       @participant = Participant.create(create_participant_params)
     end
-    @interview = Interview.find(interview_form_param)
-    @interview.participants<< (@participant)
-    redirect_to root_path
+    @interview = Interview.find(params[:participant][:interview_id])
+    
+    isValid = true
+    for i in @participant.interviews do 
+      if (interview.start_time <= i.start_time) && (i.start_time <= interview.end_time)
+        isValid = false
+      elsif (interview.start_time <= i.end_time) && (i.end_time <= interview.end_time)
+        isValid = false
+      end
+    end
+
+    if isValid
+      NewInterviewWorker.perform_async(params[:participant][:email], params[:participant][:interview_id])
+      SingleReminderWorker.perform_at(@interview.start_time - 30.minutes, @participant.email)
```

Comment:
We should add checks to prevent mail being sent in cases when interview is rescheduled

Reply:
Sir, I have handled this case in reminder_info function of "participant_mailer.rb" file.

Reply:
But `reminder_info` is not being called from `SingleReminderWorker`. How does that work?